### PR TITLE
feat: add automountserviceaccounttoken support

### LINKED ISSUE #136: add rego policy for deny automountServiceAccountToken for Pod
By default, every Pod has a service account assigned and has access to the API Server using the token created for this service account lives under the `/var/run/secrets/kubernetes.io/serviceaccount/token.` You can access the API from inside a pod using automatically mounted service account credentials.

So, it'd be worth disabling this feature by providing a Rego policy for it.

cc: @dentrax @sozercan

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>
Co-authored-by: Furkan Türkal <furkan.turkal@trendyol.com>

Fixes #136 

cc: @sozercan @dentrax

### CODE REVIEW

**Comment by sozercan** [comment]
@developer-guy  looks like tests are failing

**Comment by developer-guy** [comment]
> @developer-guy looks like tests are failing

thank you @sozercan, it should be fixed right now.

**Comment by developer-guy** [comment]
kindly reminder @sozercan

**Comment by sozercan** [comment]
do we want to check for this? https://github.com/open-policy-agent/gatekeeper-library/issues/136#issuecomment-974564456

**Comment by developer-guy** [comment]
> do we want to check for this? [#136 (comment)](https://github.com/open-policy-agent/gatekeeper-library/issues/136#issuecomment-974564456)

Exactly! Done, Thank you 🙋🏻‍♂️

File: src/general/automount-serviceaccount-token/src.rego [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	obj := input.review.object
+	mountServiceAccountToken(obj.spec)
+	msg := sprintf("Automounting service account token is disallowed, pod: %v", [obj.metadata.name])
+}
+
+mountServiceAccountToken(spec) {
+	spec.automountServiceAccountToken == true
+}
+
+# if there is no automountServiceAccountToken spec, check on volumnMount in containers. Service Account token is mounted on /var/run/secrets/kubernetes.io/serviceaccount
```
Comment (sozercan) [inline src/general/automount-serviceaccount-token/src.rego:13]: nit:
```suggestion
# if there is no automountServiceAccountToken spec, check on volumeMount in containers. Service Account token is mounted on /var/run/secrets/kubernetes.io/serviceaccount
```

**Review by sozercan — APPROVED** [review]
Thank you! LGTM

File: src/general/automount-serviceaccount-token/src.rego (hunk 1 above)
Comment (developer-guy) [reply src/general/automount-serviceaccount-token/src.rego]: thank you, done. 🙋🏻‍♂️

**Review by fseldow — APPROVED** [review]

**Review by maxsmythe — APPROVED** [review]
LGTM

Thanks for the PR!

**Pull request merged**